test/jdk/javax/swing/JFileChooser/FileSystemView/SystemIconTest.java

```diff
@@ -39,20 +38,45 @@
 
     public static void main(String[] args) {
         testSystemIcon();
-        System.out.println("ok");
+        negativeTests();
     }
 
     static void testSystemIcon() {
-        String windir = System.getenv("windir");
-        testSystemIcon(new File(windir));
-        testSystemIcon(new File(windir + "/explorer.exe"));
-        return;
+        String os = System.getProperty("os.name");
+        if (os.startsWith("Windows")) {
+            String windir = System.getenv("windir");
+            testSystemIcon(new File(windir), true);
+            testSystemIcon(new File(windir + "/explorer.exe"),
+                    true);
+        } else {
+            String homedir = System.getProperty("user.home");
+            testSystemIcon(new File(homedir), false);
+        }
+    }
+
+    static void negativeTests() {
+        ImageIcon icon;
```

Comment:
Can it be icon? This test doesn't use the fact that the returned object is `ImageIcon`.

Reply:
Fixed.